src/course_bot/internationalization.clj

```diff
@@ -0,0 +1,38 @@
+(ns course-bot.internationalization
+  (:require [clojure.string :as str])
+  (:require [taoensso.tempura :as tempura]))
+
+(def *tr-options-dict (atom {}))
+(defn add-dict [dict]
+  (swap! *tr-options-dict (partial merge-with merge) dict))
+
+(def *tr-locales (atom [:en]))
+(defn set-locales [langs]
+  (compare-and-set! *tr-locales @*tr-locales langs)
+  @*tr-locales)
+
+(defn tr [& in]
+  (let [resource (conj (apply vector in)
+                       (str "missing resource: " (str/join " " in)))]
+    (tempura/tr {:dict @*tr-options-dict}
+                @*tr-locales
+                resource)))
+
+(defn get-by-lang-and-type [lang type k] 
+  (get-in @*tr-options-dict  [lang type k]))
+
+(defn find-command [text]
+  (some (fn [[_ types]]
+          (some (fn [[_ words]]
+                  (some (fn [[k v]] (when (= v text) k)) words))
+                types))
+        @*tr-options-dict))
+
+(defn normalize-yes-no-text [text]
+  (let [inner-key (find-command (str/lower-case text))
+        lang :en
+        type :talk]
+    (if inner-key
+      (let [val (get-by-lang-and-type lang type inner-key)]
+        val)
+      "Invalid input")))
```

Comment:
I think it will be better to use `tr` as in other places. We can use `string` as a keys for tempura (if I don't mistake). Also, it will be better to reuse lang setting from the system.

Look at let-if.

I think return text as is -- better behaviour on the error, because we can simply use it chains.

Reply:
What about this problem?